episodes/chap5_python_client.md

```diff
@@ -35,75 +73,352 @@ server_api = "/api"
+| `client.store`        | Manage an algorithm store |
+| `client.algorithm`    | Manage algorithms that can be used for the computations |
+
+Almost all of these attributes provide a [get](#get), [list](#list), [create](#create),
+[update](#update) and [delete](#delete) operation. All of the functions that interact
+with the server return a dictionary with the requested information.
+
+
+::: callout
+
+### Permissions
+
+Note that the authenticated user may not be allowed to perform all operations or view all
+resources. The server will only allow the user to perform operations on the resources
+that the user has permission to perform.
 
-Each of these attributes, in turn, provides an abstraction of the CRUD (Create, Read, Update, and Delete) operations that can be performed to its corresponding concept\*:
+For example, a user may not be allowed to create a new organization, but may be allowed
+to list the organizations within all collaboration its organization participates in.
+
+:::
+
+
+### The 5 basic operations
+
+Most resources provide at least five basic operations: get, list, create, update, and
+delete.
+
+::::::::::::::::::::::::::::::::::::: tab
+
+### Get
+Get a specific *resource* with `client.<resource>.get(<id>)`. For example:
 
 ```Python
-# Get all the instances of the given '<resource>' in the server. For example, client.task.organization() returns all the organizations registered on the server.
-client.<resource>.list()
+client.organization.get(1)
+```
+```output
+{
+    'nodes': '/api/node?organization_id=1',
+    'public_key': '',
+    'studies': '/api/study?organization_id=1',
+    'name': 'Huckleberry Holdings',
+    'tasks': '/api/task?init_org_id=1',
+    'address2': '',
+    'users': '/api/user?organization_id=1',
+    'domain': 'huckleberryholdings.mc',
+    'country': 'Monaco',
+    'zipcode': '98000',
+    'runs': '/api/run?organization_id=1',
+    'address1': '4747 Huckleberry Ln',
+    'id': 1,
+    'collaborations': '/api/collaboration?organization_id=1'
+}
 ```
 
+### List
+Get all *resource* items that the user is allowed to see with
+`client.<resource>.list()`. For example:
+
 ```Python
-# Create and register a new instace of <resource> in the server.
-client.<resource>.create()
+client.organization.list()
 ```
 
-```Python
-# Delete an specific instance of the given '<resource>'.
-client.<resource>.delete()
+The output should look similar to the following:
+```output
+[
+    {'id': 1, 'name': 'Huckleberry Holdings', ...}
+    {'id': 2, 'name': 'Lychee Labs', ...},
+    {'id': 3, 'name': 'Pineapple Paradigm', ...},
+    {'id': 4, 'name': 'Huckleberry Hub', ...},
+    {'id': 5, 'name': 'Mango Matrix', ...},
+    {'id': 6, 'name': 'Apple Innovations', ...},
+    {'id': 7, 'name': 'eScience center', ...},
+    {'id': 8, 'name': 'Grapefruit Group', ...},
+    {'id': 9, 'name': 'Raspberry Revolution', ...},
+]
 ```
 
+Typically, the `list()` method returns a paginated result.
```

Comment:
Explain what a paginated result is, i.e. that you only get the first 10 results by default.
Also, I think this is always the case, so I wouldn't use 'Typically'.

Reply:
Done